pkgs/stdenv/generic/default.nix

```diff
@@ -81,7 +81,20 @@ let
 
       setup = setupScript;
 
-      inherit preHook initialPath shell
+      # We pretty much never need rpaths on Darwin, since all library path references
+      # are absolute unless we go out of our way to make them relative (like with CF)
+      # TODO: This really wants to be in stdenv/darwin but we don't have hostPlatform
+      # there (yet?) so it goes here until then.
+      preHook = preHook+ lib.optionalString buildPlatform.isDarwin ''
+        export NIX_BUILD_DONT_SET_RPATH=1
```

Comment:
I think the rpath flags for CoreFoundation are only added when this is not set. It probably should be moved outside of the conditional.

Reply:
Hmm, sorry, I don't understand. Where is the CF rpath code?

Reply:
https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/cc-wrapper/ld-wrapper.sh#L145-L147